File: CSharp/Day03/Program.cs

using Day03;
using AdventUtilities;

Dictionary<string, string> filePaths = new()
{
    ["example"] = Path.Combine(InputData.GetSolutionDirectory(), "InputFiles", "Day03", "example1.txt"),
    ["input"] = Path.Combine(InputData.GetSolutionDirectory(), "InputFiles", "Day03", "input.txt")
};

ReadOnlySpan<string> example = File.ReadAllLines(filePaths["example"]);
ReadOnlySpan<string> input = File.ReadAllLines(filePaths["input"]);
long part1 = Joltage.GetJoltage(input);
long part2 = Joltage.GetJoltage(input, partTwo: true);

Console.WriteLine($"Part 1: {part1}");
Console.WriteLine($"Part 2: {part2}");

File: FSharp/Day03/Program.fs
open AdventUtilities
open BatteryBank

let example = InputData().ReadAllLines 3 "example1"
let input = InputData().ReadAllLines 3 "input"

let part1 = checkJoltage input
let part2 = checkJoltageSequence input

printfn "Part 1: %i" part1
printfn "Part 2: %d" part2
